core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -308,54 +339,71 @@ func (o *OptimismL1Oracle) GetGasCost(ctx context.Context, tx *gethtypes.Transac
 	return assets.NewWei(l1GasCost), nil
 }
 
-func (o *OptimismL1Oracle) GetDAGasPrice(ctx context.Context) (*big.Int, error) {
-	isEcotone, err := o.checkIsEcotone(ctx)
+func (o *optimismL1Oracle) GetDAGasPrice(ctx context.Context) (*big.Int, error) {
+	err := o.checkForUpgrade(ctx)
 	if err != nil {
 		return nil, err
 	}
-
-	o.logger.Infof("Chain isEcotone result: %t", isEcotone)
-
-	if isEcotone {
-		return o.getEcotoneGasPrice(ctx)
+	if o.isFjord || o.isEcotone {
+		return o.getEcotoneFjordGasPrice(ctx)
 	}
 
 	return o.getV1GasPrice(ctx)
 }
 
-func (o *OptimismL1Oracle) checkIsEcotone(ctx context.Context) (bool, error) {
-	// if chain is already Ecotone, NOOP
-	if o.isEcotone {
-		return true, nil
+// Checks oracle flags for Ecotone and Fjord upgrades
+func (o *optimismL1Oracle) checkForUpgrade(ctx context.Context) error {
+	// if chain is already Fjord (the latest upgrade), NOOP
+	// need to continue to check if not on latest upgrade
+	if o.isFjord {
```

Comment:
Don't we want to check for both?
```suggestion
	if o.isFjord || o.isEcotone {
```

Reply:
If we checked for both, we'd exit out early if the Ecotone upgrade is live. We wouldn't want to do this since we'd stop checking for the Fjord upgrade which is after. So we only want to exit early if the latest upgrade has already occurred.

Reply:
But Ecotone and Fjord have the same calculation formula, don't they?

Reply:
In that case, I don't think we'd even need the `isFjord` flag. We would never make the RPC call to update it with this change. My worry would be if we later need to add some Fjord specific logic this would not be as obvious for whoever makes the change if they trust `checkForUpgrade` to work as expected.